Source/Android/jni/MainAndroid.cpp

```diff
@@ -602,6 +605,75 @@ JNIEXPORT void JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_eglBindAPI(J
   eglBindAPI(api);
 }
 
+JNIEXPORT void JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_InitGameIni(
+JNIEnv* env, jobject obj, jobject assetManager, jstring jGameID)
+{
+  IniFile ini;
+  std::string gameid = GetJString(env, jGameID);
+
+  std::vector<std::string> cfgnames = ConfigLoaders::GetGameIniFilenames(gameid, 0);
+
+  std::string baseName = cfgnames[0];
+
+  AAssetManager* mgr = AAssetManager_fromJava(env, assetManager);
+
+  // Grab the underlying config files for a given game ID, using the asset manager passed from Java to pull them out of the APK
+  for (const std::string& filename : cfgnames ) {
+    std::string fullpath = "GameSettings/" + filename;
+    AAsset* asset = AAssetManager_open(mgr, fullpath.c_str(), AASSET_MODE_UNKNOWN);
+    if (NULL == asset) {
```

Comment:
We don't use inverted conditionals like this (or `NULL`, `nullptr` is preferable), either `!asset` or `asset == nullptr` should be used instead.